src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/LanguageServices/ActiveConfiguredProjectsIgnoringTargetFrameworkProvider.cs

```diff
@@ -0,0 +1,180 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using Microsoft.VisualStudio.ProjectSystem.Build;
+using System;
+using System.Collections.Immutable;
+using System.ComponentModel.Composition;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Threading.Tasks.Dataflow;
+
+namespace Microsoft.VisualStudio.ProjectSystem.LanguageServices
+{
+    /// <summary>
+    ///     Provides set of configured projects whose ProjectConfiguration has all dimensions (ignoring the TargetFramework) matching the active VS configuration.
+    ///
+    ///     For example, for a cross-targeting project with TargetFrameworks = "net45;net46" we have:
+    ///     -> All known configurations:
+    ///         Debug | AnyCPU | net45
+    ///         Debug | AnyCPU | net46
+    ///         Release | AnyCPU | net45
+    ///         Release | AnyCPU | net46
+    ///         
+    ///     -> Say, active VS configuration = "Debug | AnyCPU"
+    ///       
+    ///     -> Active configurations ignoring TargetFramework returned by this provider:
+    ///         Debug | AnyCPU | net45
+    ///         Debug | AnyCPU | net46
+    /// </summary>
+    [Export(typeof(ActiveConfiguredProjectsIgnoringTargetFrameworkProvider))]
+    internal class ActiveConfiguredProjectsIgnoringTargetFrameworkProvider : OnceInitializedOnceDisposedAsync
```

Comment:
This name doesn't make sense to both as a exchange (interface) and implementation. You're not ignoring target frameworks, you're factoring them in your decision to determine what the active configs are.

The exchange type in this, is "active configured projects". I think we should extract out an interface and call it that. It will be a pain to test this class, as you need to setup a bunch of state.